StatisticsInR/2.1_simple_linear_regression.Rmd

```diff
@@ -0,0 +1,1223 @@
+
+```{r, echo = T, warning = F}
+predict(DiveModel, newdata=data.frame(BodySizeKG=75000), 
+                             type="response", interval = "prediction")
+```
+
+</br>
+
+The mean prediction is 352m deep. This seems ok. But when we look at the prediction interval, we see that when we include uncertainty, we are not even sure if they whale will dive below the surface by 2km or jump into the air by 1.3km. When we include uncertainty, it is clear that based on the current data and model, we cannot say anything about the possible dive depth of the new whale. We even get biologically unrealistic predictions. 
+
+</br>
+
+This is something we will look at in the next section.
+
+</br>
+
+## <i class="fas fa-tasks"></i> Model checking {.tabset .tabset-fade}
+
+We have now have a model, estimates of parameters, and have calculated the uncertainty of the parameters. **But how do we know if this model is any good?** 
+
+</br>
+
+### Theory
+
+</br>
+
+To find out if our model is any good from a theoretical perspective, we need to check if the model meets the five assumptions of the linear regression that are stated in the **Model details** section above. 
+
+</br>
+
+For this, we can use graphs called **diagnostic plots**. There are four key diagnostic plots that we use for simple linear regression and each plot tests whether a different assumption has been met. To make the diagnostic plots, we use the `plot()` function with
+our linear model object as the first argument and `which = number` as the second. The number
+should be replaced by the number corresponding to the plot you want. 
+
+</br>
+
+For more on what each plot means, go here: [Model checking](). On this page, we will look at some example plots for a simple linear regression. 
+
+#### 1. Residuals vs fitted plot
+
+</br>
+
+There are two examples of residuals vs fitted plots shown below. If our assumptions
+are met, we expect a few characteristics of the plot:
+
+* The red line should be horizontally straight and at 0.
+* There should be no structure in the residuals.
+* The residuals should not curve.
+
+</br>
+
+```{r, echo=F, warning=F, message=F}
+set.seed(2020)
+ExampleData$a <- rpois(100,as.numeric(ExampleData$z)^1.2)
+
+par(mfrow=c(1,2))
+
+ExampleModel1 <- lm(x ~ y, data = ExampleData)
+plot(ExampleModel1, which = 1)
+title(main="Good", line = 1)
+
+ExampleModel2 <- lm(y ~ a, data = ExampleData)
+plot(ExampleModel2, which = 1)
+title(main="Bad", line = 1)
+```
+*Figure 10: Example of a good (left) and a bad (right) residuals vs fitted plot for a simple linear regression.*
+
+**In reality, it will not be as clear cut as the examples above. You will need to use
+your own judgement to decide if the assumption is sufficiently met. Do not expect perfection.**
+
+</br>
+
+#### Normal QQ plot
+
+There are two examples of normal QQ plots shown below. If our assumptions
+are met, we expect:
+
+* The points lie along the line.
+
```

Comment:
Maybe also point out for all of the assumptions that the line/plot is usually not perfect for "biological" data. Like explain what is still an acceptable deviation from the 1:1 line. This is often not clear for student, and it is to some extent subjective.

Reply:
I tried to use more realistic examples and put disclaimers under each figure to stress this. Was it not clear enough? I agree it is an important point to get across so it isn't clear I should fix it.

Reply:
I cannot see the full example right now, but using real biological data or messy data helps, to not get the line perfect. And maybe just have a short discussion when is a model "good enough". But maybe you already have this. I can have a look once you have finalised this page.

Reply:
Great